draft-ietf-tls-esni.md

```diff
@@ -1252,16 +1238,83 @@ TCP connections an attacker can open.
 
 ### Do Not Stick Out {#dont-stick-out}
 
-The only explicit signal indicating possible use of ECH is the ClientHello
-"encrypted_client_hello" extension. Server handshake messages do not contain any
-signal indicating use or negotiation of ECH. Clients MAY GREASE the
-"encrypted_client_hello" extension, as described in {{grease-ech}}, which helps
-ensure the ecosystem handles ECH correctly. Moreover, as more clients enable ECH
-support, e.g., as normal part of Web browser functionality, with keys supplied
-by shared hosting providers, the presence of ECH extensions becomes less unusual
-and part of typical client behavior. In other words, if all Web browsers start
-using ECH, the presence of this value will not signal unusual behavior to
-passive eavesdroppers.
+As a means of reducing the impact of network ossification, {{?RFC8744}}
+recommends SNI-protection mechanisms be designed in such a way that network
+operators do not differentiate connections using the mechanism from connections
+not using the mechanism. To that end, ECH is designed to resemble a standard
+TLS handshake as much as possible. The most obvious difference is the extension
+itself: as long as middleboxes ignore it, as required by {{!RFC8446}}, the rest
+of the handshake is designed to look very much as usual.
+
+The GREASE ECH protocol described in {{grease-ech}} provides a low-risk way to
+evaluate the deployability of ECH. It is designed to mimic the real ECH protocol
+({{real-ech}}) without changing the security properties of the handshake. The
+underlying theory is that if GREASE ECH is deployable without triggering
+udifferentiable behavior, and real ECH is indifferentiable from GREASE ECH, then
+ECH should be deployable as well. Thus, our strategy for mitigating network
+ossification has two parts:
+1. Design a "cover" protocol (i.e., GREASE ECH) that meets some notion of
+   deployability.
+1. Design the "target" protocol (i.e., ECH) so that networks do not differentiate
+   between it and the cover protocol.
+
+This strategy is analogous to the methodology that lead to "Middlebox
+Compatibility Mode" for TLS 1.3 described in {{!RFC8446}}.
+
+In the first step, the deployability of GREASE ECH is established by deploying
+the protocol and measuring its impact. [[OPEN ISSUE: Run an experiment that
```

Comment:
Is this an open issue in the sense that the data from that experiment needs to be in this draft? If not, I'm not sure it should be mentioned here.

Reply:
I think the point is that we will first run an experiment -- before this document is published as an RFC -- and then update this text with the results. Perhaps this helps?

```suggestion
the protocol and measuring its impact. [[TODO: Document results from preliminary experiment that
```

Reply:
Yup, that's what I mean!